### PR TITLE
Create Event Feature: Views Testing

### LINKED ISSUE #70: Adding Views Test To CreateEvent Feature

### PR DESCRIPTION
**Pull request by RavidYael**
All testing scenarios were tested in create_event.py (form testing) - now tested via view layer.
fix: #70

### CODE REVIEW

**Comment by RavidYael** [comment]
@MortarDefender, @danninon, I believe there is a problem with your [conftest.py](https://github.com/redhat-beyond/lets-meet/blob/main/reminders/conftest.py) file in the reminders app. 
between lines 26 to 31 you define the start_time to be 15:00 and the end_time to be 13:00 what causes some bugs as it affects the pytest configuration. 
Can you please help with that?

File: events/conftest.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    pytest.invalid_date_time = datetime(
+        pytest.current_date.year - 1, pytest.current_date.month, pytest.current_date.day, 12, 0, 0, tzinfo=timezone.utc
+    )
+
+    pytest.date_time_end = datetime(
+        pytest.current_date.year, pytest.current_date.month, pytest.current_date.day + 1, 14, 0, 0, tzinfo=timezone.utc
+    )
+    pytest.date_time_start = datetime(
+        pytest.current_date.year, pytest.current_date.month, pytest.current_date.day + 1, 13, 0, 0, tzinfo=timezone.utc
+    )
```
Comment (MortarDefender) [inline events/conftest.py]:
```suggestion
    pytest.event_date_time_end = datetime(
        pytest.current_date.year, pytest.current_date.month, pytest.current_date.day + 1, 14, 0, 0, tzinfo=timezone.utc
    )
    pytest.event_date_time_start = datetime(
        pytest.current_date.year, pytest.current_date.month, pytest.current_date.day + 1, 13, 0, 0, tzinfo=timezone.utc
    )
```

File: events/conftest.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+@pytest.fixture
+def sign_in(client, signed_up_user_details):
+    return client.post('/login/', data=signed_up_user_details)
+
+
+@pytest.fixture
+def valid_event_data_and_reminder():
+    return {'title': TITLE, 'date_time_start': pytest.date_time_start,
+            'date_time_end': pytest.date_time_end, 'color': Colors.BLACK,
```
Comment (MortarDefender) [inline events/conftest.py]:
```suggestion
    return {'title': TITLE, 'date_time_start': pytest.event_date_time_start,
            'date_time_end': pytest.event_date_time_end, 'color': Colors.BLACK,
```
```suggestion
    return {'title': TITLE, 'date_time_start': pytest.date_time_start,
            'date_time_end': pytest.date_time_end, 'color': Colors.BLACK,
```

Comment (MortarDefender) [inline events/conftest.py:53]:
```suggestion
    return {'title': TITLE, 'date_time_start': pytest.event_date_time_start,
            'date_time_end': pytest.event_date_time_end, 'color': Colors.BLACK,
```
```suggestion
    return {'title': TITLE, 'date_time_start': pytest.date_time_start,
            'date_time_end': pytest.date_time_end, 'color': Colors.BLACK,
```

**Review by MortarDefender — CHANGES_REQUESTED** [review]
When you are using pytest variables like the variables in pytest configure
the variables are saved in pytest memory until the end of the tests,
so in your case there is another conftest file in the reminders app that redefine date_time_start.
I have change it to event_date_time start/end to distinguish between them.

File: events/conftest.py (hunk 1 above)
Comment (RavidYael) [reply events/conftest.py]: Yeah thats another way to solve that, tnx

**Comment by RavidYael** [comment]
> When you are using pytest variables like the variables in pytest configure the variables are saved in pytest memory until the end of the tests, so in your case there is another conftest file in the reminders app that redefine date_time_start. I have change it to event_date_time start/end to distinguish between them.

File: events/conftest.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    pytest.invalid_method = "undefined"
+    pytest.current_date = datetime.now()
+    pytest.valid_method = ReminderType.EMAIL
+    pytest.message = "Joined Meeting in 50 minutes"
+    pytest.valid_date_time = datetime(
+        pytest.current_date.year, pytest.current_date.month, pytest.current_date.day + 1, 12, 0, 0, tzinfo=timezone.utc
+    )
+    pytest.invalid_date_time = datetime(
+        pytest.current_date.year - 1, pytest.current_date.month, pytest.current_date.day, 12, 0, 0, tzinfo=timezone.utc
+    )
```
Comment (MortarDefender) [inline events/conftest.py]: These variables are not relevant to your tests and are not there for used by your functions
Please remove them from this PR,
In a later PR we will add conftest to all the apps and create the relevant variables if needed

File: events/class_tests/test_create_event_views.py [hunk 4]
```diff
@@ -0,0 +1,63 @@
+import pytest
+from pytest_django.asserts import assertTemplateUsed
+from events.class_tests.create_events import DATE_TIME_START, DATE_TIME_END, COLOR, TITLE
+from reminders.class_models.reminder_models import ReminderType
+from events.models import Event, Colors
+
+CREATE_EVENT_URL = "/event/create/"
+CREATE_EVENT_TEMPLATE = 'events/create_event.html'
+LOGIN_PAGE_TEMPLATE = 'login/register_login.html'
```
Comment (MortarDefender) [inline events/class_tests/test_create_event_views.py]: Please add these variables to the conftest.

An example of this is
https://github.com/redhat-beyond/lets-meet/blob/830988ce721fc82eafb792293b51bb2145b6de79/reminders/class_tests/create_reminder_tests.py#L54-L57

and here is the definition:
https://github.com/redhat-beyond/lets-meet/blob/830988ce721fc82eafb792293b51bb2145b6de79/reminders/conftest.py#L36-L38

File: events/class_tests/test_create_event_views.py [hunk 5]
```diff
@@ -0,0 +1,63 @@
+import pytest
+from pytest_django.asserts import assertTemplateUsed
+from events.class_tests.create_events import DATE_TIME_START, DATE_TIME_END, COLOR, TITLE
```
Comment (MortarDefender) [inline events/class_tests/test_create_event_views.py]: Please use only the conftest variables, the same applies to every line of code you write with constant variable that you imported or have written yourself.
if you have added a conftest file there is a better way to represent constant variable as you can see here:
https://github.com/redhat-beyond/lets-meet/blob/830988ce721fc82eafb792293b51bb2145b6de79/reminders/conftest.py#L15-L18

**Review by MortarDefender — CHANGES_REQUESTED** [review]

File: events/conftest.py (hunk 3 above)
Comment (RavidYael) [reply events/conftest.py]: Actually, these variables are in use in my tests.

File: events/class_tests/test_create_event_views.py (hunk 5 above)
Comment (RavidYael) [reply events/class_tests/test_create_event_views.py]: Done

File: events/class_tests/test_create_event_views.py (hunk 4 above)
Comment (RavidYael) [reply events/class_tests/test_create_event_views.py]: Done

File: events/conftest.py [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+
+    pytest.CREATE_EVENT_URL = "/event/create/"
+    pytest.CREATE_EVENT_TEMPLATE = 'events/create_event.html'
+    pytest.LOGIN_PAGE_TEMPLATE = 'login/register_login.html'
+
+    pytest.valid_event_date_time_start = DATE_TIME_START
+    pytest.valid_event_date_time_end = DATE_TIME_END
+    pytest.valid_event_color = COLOR
+    pytest.valid_event_title = TITLE
```
Comment (MortarDefender) [inline events/conftest.py]: Hi @RavidYael 
can you copy lines 7 - 12 into lines 31 - 34
lines 31 - 34 are designed to replace the constant like lines 7 - 12.
And please use lower case letters in this variables 😄

File: events/conftest.py [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+@pytest.fixture
+def sign_in(client, signed_up_user_details):
+    return client.post('/login/', data=signed_up_user_details)
+
+
+@pytest.fixture
+def valid_event_data_and_reminder():
+    return {'title': TITLE, 'date_time_start': pytest.event_date_time_start,
+            'date_time_end': pytest.event_date_time_end, 'color': Colors.BLACK,
+            'date_time': pytest.reminder_date_time, 'method': ReminderType.EMAIL}
```
Comment (MortarDefender) [inline events/conftest.py]: Please use pytest consts as you declared above.

I mean: 
Title => pytest.title

File: events/class_tests/test_create_event_views.py [hunk 8]
```diff
@@ -0,0 +1,58 @@
+import pytest
```
Comment (MortarDefender) [inline events/class_tests/test_create_event_views.py:1]: Can you change the name of this file to:
"create_event_views_test.py"

**Review by MortarDefender — CHANGES_REQUESTED** [review]

File: events/conftest.py (hunk 6 above)
Comment (RavidYael) [reply events/conftest.py:34]: about 'can you copy lines 7 - 12 into lines 31 - 34':
I believe the way it is written right now improves the maintainability of the code, is there any technical reason you suggest that?
about the lower case - done.

File: events/conftest.py (hunk 7 above)
Comment (RavidYael) [reply events/conftest.py]: done

File: events/class_tests/test_create_event_views.py (hunk 8 above)
Comment (RavidYael) [reply events/class_tests/test_create_event_views.py:1]: done

File: events/conftest.py (hunk 6 above)
Comment (RavidYael) [reply events/conftest.py]: done

**Review by MortarDefender — APPROVED** [review]
Nicely Done 😄

**Review by RonHachmon — APPROVED** [review]
LGTM

**Review by YaelD — APPROVED** [review]
LGTM

**Review by kobihk — APPROVED** [review]

**Pull request merged**